### ISSUE TITLE
[v.1.12.0] Release Tracker

### ISSUE THREAD

**Issue by atalman** [pytorch/pytorch#78005]
### 🚀 The feature, motivation and pitch

We cut a [release branch](https://github.com/pytorch/pytorch/tree/release/1.12) for the 1.12.0 release.

Our plan from this point from this point is roughly:

* Phase 1 (until 06/13): work on finalizing the release branch
* Phase 2 (after 06/13): perform extended integration/stability/performance testing based on Release Candidate builds.

This issue is for tracking cherry-picks to the release branch.

## Cherry-Pick Criteria

**Phase 1 (until 06/13/22):**

Only low-risk changes may be cherry-picked from master:

1. Fixes to regressions against the most recent minor release (e.g. 1.11 for 1.12 release; see [module: regression issue list](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22module%3A+regression%22+))
2. Critical fixes for: [silent correctness](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+correctness+%28silent%29%22), [backwards compatibility](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+bc-breaking%22+), [crashes](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+crash%22+), [deadlocks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+deadlock%22+), (large) [memory leaks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+memory+usage%22+)
3. Fixes to new features introduced in the most recent minor release (e.g. 1.11.0 for 1.12 release)
4. Test/CI fixes
5. Documentation improvements
6. Compilation fixes or ifdefs required for different versions of the compilers or third-party libraries
7. Release branch specific changes (e.g. change version identifiers)

Any other change requires special dispensation from the release managers (currently @atalman, @seemethere ). If this applies to your change please write "Special Dispensation" in the "Criteria Category:" template below and explain.

**Phase 2 (after 06/13/22):**

Note that changes here require us to rebuild a Release Candidate and restart extended testing (likely delaying the release). Therefore, the only accepted changes are **Release-blocking** critical fixes for: [silent correctness](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+correctness+%28silent%29%22), [backwards compatibility](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+bc-breaking%22+), [crashes](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+crash%22+), [deadlocks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+deadlock%22+), (large) [memory leaks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+memory+usage%22+)

Changes will likely require a discussion with the larger release team over VC or Slack.

## Cherry-Pick Process

1. Ensure your PR has landed in master. This does not apply for release-branch specific changes (see Phase 1 criteria).
2. Create (but do not land) a PR against the [release branch](https://github.com/pytorch/pytorch/tree/release/1.12).
   <details>

    ```bash
    # Find the hash of the commit you want to cherry pick
    # (for example, abcdef12345)
    git log

    git fetch origin release/1.11
    git checkout release/1.11
    git cherry-pick abcdef12345

    # Submit a PR based against 'release/1.10' either:
    # via the GitHub UI
    git push my-fork

    # via the GitHub CLI
    gh pr create --base release/1.11
    ```

    </details>
3. Make a request below with the following format:

```
Link to landed master PR (if applicable):
* 

Link to release branch PR:
* 

Criteria Category:
* 
```

1. Someone from the release team will reply with approved / denied or ask for more information.
2. If approved, someone from the release team will merge your PR once the tests pass. **Do not land the release branch PR yourself.**

**NOTE: Our normal tools (ghstack / ghimport, etc.) do not work on the release branch.**

Please note HUD Link with branch CI status and link to the HUD to be provided here.
[HUD](https://hud.pytorch.org/hud/pytorch/pytorch/release%2F1.12)

### Versions

1.12.0

**Comment by zou3519** [comment]
Link to landed master PR (if applicable):
* n/a

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/78021

Criteria Category:
* Release branch specific changes (removing prototype feature that should not be in release)

**Comment by pritamdamania87** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/77987

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/78056

Criteria Category:
* Test/CI fixes

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
NA

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/78174

Criteria Category:
Critical CI fixe

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/77964

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/78198

Criteria Category:
* Critical Fixes

**Comment by zou3519** [comment]
Link to landed master PR (if applicable):
- n/a

Link to release branch PR:
- https://github.com/pytorch/pytorch/pull/78300

Criteria Category:
- Release branch specific changes (removing incomplete feature from release)

**Comment by JackCaoG** [comment]
Link to release branch PR:

https://github.com/pytorch/pytorch/pull/77993

Criteria Category:
updating pytorh/xla 1.12 branch for CI testing.

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/78085

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/78370

Criteria Category:
* Critical Fix

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/builder/pull/1040

Link to release branch PR:
* 

Criteria Category:
* Critical CI fix

**Comment by kshitij12345** [comment]
Link to landed master PR (if applicable):
*  https://github.com/pytorch/pytorch/pull/78351

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/78709

Criteria Category: Documentation improvements

cc: @anjali411

**Comment by NivekT** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/78673

Link to release branch PR: 
* https://github.com/pytorch/pytorch/pull/78726

Criteria Category: 
* Fixes to regressions - we introduced a change that causes noticeable performance issue in the data loading process (specifically `IterDataPipe`). This cherry-pick addresses that issue.

**Comment by yuguo68** [comment]
Link to landed master PR (if applicable):
https://github.com/pytorch/pytorch/pull/78246

Link to release branch PR:
https://github.com/pytorch/pytorch/pull/78746

Criteria Category:
silent correctness, one liner fix to a missing gradient info issue

**Comment by VitalyFedyunin** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/78631

Link to release branch PR:
*  https://github.com/pytorch/pytorch/pull/78762

Criteria Category:
*  DataLoader backward compatibility correctness

**Comment by kshitij12345** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/78245

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/78710

Criteria Category:
* `chalf` support is still experimental. The PR warns user when creating `chalf` tensor.

cc: @anjali411

**Comment by mrshenli** [comment]
Link to landed master PR (if applicable):

* https://github.com/pytorch/pytorch/pull/78666

Link to release branch PR:

* https://github.com/pytorch/pytorch/pull/78810

Criteria Category:

Critical fixes for: backwards compatibility. See #78549

**Comment by ejguan** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/78765

Link to release branch PR:
*  https://github.com/pytorch/pytorch/pull/78927

Criteria Category:
*  DataLoader backward compatibility correctness

**Comment by NivekT** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/78419

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/78948

Criteria Category:
* Fixes to new features introduced in the most recent minor release - we introduced new functional API names for various DataPipes in `torchdata`, and this batch correctly aligns the API names.

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/78230

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/78962

Criteria Category:
* MPS Fix

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/78408

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/78963

Criteria Category:
* MPS Fix

**Comment by VitalyFedyunin** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/78950

Link to release branch PR:
*  https://github.com/pytorch/pytorch/pull/78967

Criteria Category:
*  DataLoader backward compatibility correctness

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* 

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/78974

Criteria Category:
* MPS Feature fixes

**Comment by awgu** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/78784

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/78981

Criteria Category:
* Documentation improvements

**Comment by NivekT** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/78674
* https://github.com/pytorch/pytorch/pull/78970

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/79027

Criteria Category:
* Fixes to performance regressions
* Release branch specific changes (deprecation warnings)

**Comment by erichan1** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/78832

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/79029

Criteria Category:
* Critical fix (for torch.nn.TransformerEncoder and torchtext)

**Comment by ejguan** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/79041

Link to release branch PR:
*  https://github.com/pytorch/pytorch/pull/79063

Criteria Category:
*  Fixes to new features introduced in the most recent minor release (e.g. 1.11.0 for 1.12 release)

**Comment by zou3519** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/78942

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/79123

Criteria Category:
* Documentation improvement; Critical fix for "half-complete feature"
* For more context: torchgen is a package for PyTorch code generation. Unfortunately, with the branch cut, torchgen is in an incomplete state AND is being packaged with the pytorch binaries. We decided to complete the feature rather than remove it, because it is difficult to remove (the rest of PyTorch depends on it). This is a low-risk change that involves copying some additional files into the torchgen folder and adding documentation.

**Comment by ejguan** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/79124

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/79129

Criteria Category:
*  Fixes to new features introduced in the most recent minor release (e.g. 1.11.0 for 1.12 release)

**Comment by eqy** [comment]
Link to landed master PR (if applicable):

https://github.com/pytorch/pytorch/pull/79060
Link to release branch PR:

https://github.com/pytorch/pytorch/pull/79071

CC @ptrblck @ngimel

**Comment by yuguo68** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/79101

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/79249

Criteria Category:
* silent correctness, fix a jit wrong operator schema issue

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
*  https://github.com/pytorch/builder/pull/1055

Link to release branch PR:
* https://github.com/pytorch/builder/pull/1058

Criteria Category:
* Critical FIX

**Comment by NivekT** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/79302

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/79309

Criteria Category:
* Typo fix to regressions against the most recent minor release

**Comment by malfet** [comment]
Link to landed master PR:

 * #78575

Link to release branch PR:

 * https://github.com/pytorch/pytorch/pull/79315

Criteria Category:

  * Fixes regression from previous release
---
@malfet: merged

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/79220

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/79416

Criteria Category:
*  Critical CI Fixes

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/builder/pull/1012

Link to release branch PR:
* https://github.com/pytorch/builder/commit/72c069781f7882f60a3355765890a4c2e686d071

Criteria Category:
* Critical CI fixes

---
@malfet merged

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/79024

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/79429

Criteria Category:
* Critical CI Fixes

**Comment by ngimel** [comment]
Landed master PR #79401
Cherry pick #79455
Criteria: documentation improvements

**Comment by mcarilli** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/77862

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/79472

Criteria Category:
* Critical fix (without this PR, CUDA graph capture of Adam and AdamW produce silently incorrect results)

**Comment by anjali411** [comment]
Link to landed master PR: https://github.com/pytorch/pytorch/pull/78753

Link to release branch PR: https://github.com/pytorch/pytorch/pull/79481

Criteria Category: docs

**Comment by rohan-varma** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/78752

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/79490

Criteria Category:
* Critical fix (silent correctness)

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/79324

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/79522

Criteria Category:
* Critical CI Fix

**Comment by atalman** [comment]
Please note we are currently in Phase 2 (after 06/13): perform extended integration/stability/performance testing based on Release Candidate builds. Therefore, the only accepted changes are Release-blocking critical fixes for: [silent correctness](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+correctness+%28silent%29%22), [backwards compatibility](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+bc-breaking%22+), [crashes](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+crash%22+), [deadlocks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+deadlock%22+), (large) [memory leaks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+memory+usage%22+)

**Comment by eqy** [comment]
Link to landed master PR (if applicable):

+ https://github.com/pytorch/pytorch/pull/79537
Link to release branch PR:

+ https://github.com/pytorch/pytorch/pull/79539
Criteria Category:

documentation improvements

**Comment by ejguan** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/79524

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/79550

Criteria Category:
* Critical fix (silent correctness)
* See issue: https://github.com/pytorch/pytorch/issues/79449

**Comment by malfet** [comment]
> Link to landed master PR (if applicable):
> 
> * [Enable NNPACK for MacOS binaries builder#1055](https://github.com/pytorch/builder/pull/1055)
> 
> Link to release branch PR:
> 
> * [Enable NNPACK for MacOS binaries (#1055) builder#1058](https://github.com/pytorch/builder/pull/1058)
> 
> Criteria Category:
> 
> * Critical FIX
> 

Cherry-picked https://github.com/pytorch/pytorch/pull/78326  (as https://github.com/pytorch/pytorch/commit/1d5ecdb3b913b03273ff2477822719b36679eb2b) into release 1.12, which should have been done as related change

**Comment by jbschlosser** [comment]
Link to landed master PRs: #79545 and #79549
Link to release branch PRs: #79610 and #79611
Criteria Category: Critical fix for silent BC-breaking behavior

**Comment by erichan1** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/79480
Link to release branch PR:
* TODO - master PR has not been landed yet, although asap (by EOD 6/16)
Criteria Category:
* Critical fix for Torchscript + nn.Transformer being broken
* I understand that this is late - @atalman let me know if we can get this in.

**Comment by ejguan** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/79829
Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/79890
Criteria Category:
* Critical fix for TorchData + PyTorch Distributed with NCCL backend


cc: @atalman

**Comment by seemethere** [comment]
Link to landed master PR (if applicable):
* n/a

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/79884

Criteria Category:
* [backwards compatibility](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+bc-breaking%22+sort%3Aupdated-desc)

**Comment by malfet** [comment]
Link to landed master PR:

* https://github.com/pytorch/pytorch/pull/79865

Link to release branch PR:
* https://github.com/pytorch/pytorch/commit/50090861500adfd7da975a875aca0df636956df4

Criteria Category:
* Critical infra fixes: restores tagged doc builds on release branch

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/pytorch/pull/79929

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/79983

Criteria Category:
* Critical CI fix

**Comment by atalman** [comment]
Release is completed, closing this
